### PR TITLE
refactor(core)!: make the default value of `retry` and `repeats` 0

### PR DESCRIPTION
**Pull request by Dunqing**
close: #3631

### CODE REVIEW

File: packages/runner/src/run.ts [hunk 1]
```diff
@@ -132,12 +132,10 @@ export async function runTest(test: Test, runner: VitestRunner) {
...
 
-  const repeats = typeof test.repeats === 'number' ? test.repeats : 1
+  const { repeats, retry } = test
 
-  for (let repeatCount = 0; repeatCount < repeats; repeatCount++) {
-    const retry = test.retry || 1
-
-    for (let retryCount = 0; retryCount < retry; retryCount++) {
+  for (let repeatCount = 0; repeatCount <= repeats; repeatCount++) {
+    for (let retryCount = 0; retryCount <= retry; retryCount++) {
```
Comment (sheremet-va) [inline packages/runner/src/run.ts:138]: There is still an issue that retry count resets with each repeat

**Review by sheremet-va — CHANGES_REQUESTED** [review]
I think we can also add a test that calls startVitest and checks retry count on vitest.state

File: packages/runner/src/run.ts (hunk 1 above)
Comment (Dunqing) [reply packages/runner/src/run.ts:138]: In my opinion, I think repeat should reset the retry count, repeat should have a separate retry count.

Comment (Dunqing) [reply packages/runner/src/run.ts:138]: Repeat does not make the test stable, when a test is bound to fail once or twice, then if he uses repeat the test will not succeed

Comment (sheremet-va) [reply packages/runner/src/run.ts:138]: Repeat is not for stability, retry is. We lose all information about previous retries which is bad. We should either store it differently or increment it each time

Comment (Dunqing) [reply packages/runner/src/run.ts:138]: "increment" sounds good

Comment (Dunqing) [reply packages/runner/src/run.ts:138]: image

Now retryCount will increase with each repeat.

**Review by sheremet-va — APPROVED** [review]

**Comment by sheremet-va** [comment]
I don't think this is a breaking change, it's publically available only with `createVitest` and it's not subjected to semver

**Pull request merged**